contracts/interchain-token-service/src/primitives.rs

```diff
@@ -29,6 +30,10 @@ pub enum Message {
     InterchainTransfer(InterchainTransfer),
     /// Deploy a new interchain token on the destination chain
     DeployInterchainToken(DeployInterchainToken),
+
+    RegisterToken(RegisterTokenMetadata),
```

Comment:
```suggestion
    RegisterTokenMetadata(RegisterTokenMetadata),
```
Rename message type as well